refactor/conv_modules.py

```diff
@@ -28,57 +28,52 @@ def __init__(
         groups: int = 1,
         bias: bool = True,
         padding_mode: str = 'zeros',  # TODO: refine this type
-        device=torch.device('cuda' if torch.cuda.is_available() else 'cpu'),
+        device=None,
         dtype=None
     ) -> None:
         # init as Conv2d
         super().__init__(
             in_channels, out_channels, kernel_size, stride, padding, dilation, 
             groups, bias, padding_mode, device, dtype)
-
-        factory_kwargs = {'device': device, 'dtype': dtype}
-        self.factory_kwargs = factory_kwargs
-
         # weight shape: (out_channels, in_channels // groups, *kernel_size)
         weight_shape = self.weight.shape
+        new_weight = self.weight.new_empty(weight_shape)
         del self.weight # remove Parameter, create buffer
-        self.register_buffer("weight", torch.empty(weight_shape, **factory_kwargs))
+        self.register_buffer("weight", new_weight)
```

Comment:
[Why not `torch.empty_like()`?](http://pytorch.org/docs/master/generated/torch.empty_like.html?highlight=empty_like#torch.empty_like)

Reply:
honestly thats better yeah.